packages/plugins/response-cache-cloudflare-kv/src/invalidate.ts

```diff
@@ -0,0 +1,44 @@
+import type { CacheEntityRecord } from '@envelop/response-cache';
+import type { KvCacheConfig } from '.';
+import { buildEntityKey } from './cache-key';
+
+export async function invalidate(
+  entities: Iterable<CacheEntityRecord>,
+  config: KvCacheConfig,
+): Promise<void> {
+  const kvPromises: Promise<unknown>[] = []; // Collecting all the KV operations so we can await them all at once
+
+  for (const entity of entities) {
+    const entityKey = buildEntityKey(entity.typename, entity.id, config.keyPrefix);
+
+    for await (const kvKey of _getAllKvKeysForPrefix(entityKey, config)) {
+      if (kvKey.metadata?.operationKey) {
+        kvPromises.push(config.KV.delete(kvKey.metadata?.operationKey));
+        kvPromises.push(config.KV.delete(kvKey.name));
+      }
+    }
+  }
+
+  await Promise.allSettled(kvPromises);
+}
+
+export async function* _getAllKvKeysForPrefix(prefix: string, config: KvCacheConfig) {
```

Comment:
We generally don't use `_` prefix for functions. A function should be either considered private and not exported or public and exported. I think it is ok to have it exported without the `_` prefix :-)